test/test_blocks_on_demand.py

```diff
@@ -0,0 +1,106 @@
+"""
+Test blocks on demand mode.
+"""
+
+from test.rpc.rpc_utils import gateway_call
+
+import requests
+
+from .account import invoke
+from .settings import APP_URL
+from .shared import (
+    ABI_PATH,
+    CONTRACT_PATH,
+    PREDEPLOY_ACCOUNT_CLI_ARGS,
+    PREDEPLOYED_ACCOUNT_ADDRESS,
+    PREDEPLOYED_ACCOUNT_PRIVATE_KEY,
+)
+from .util import (
+    ReturnCodeAssertionError,
+    assert_tx_status,
+    call,
+    deploy,
+    devnet_in_background,
+)
+
+
+@devnet_in_background(*PREDEPLOY_ACCOUNT_CLI_ARGS, "--blocks-on-demand")
+def test_blocks_on_demand_invoke():
+    """Test deploy in blocks-on-demand mode"""
+    latest_block = gateway_call("get_block", blockNumber="latest")
+    genesis_block_number = latest_block["block_number"]
+    assert genesis_block_number == 0
+
+    deploy_info = deploy(CONTRACT_PATH, inputs=["0"])
+    assert_tx_status(deploy_info["tx_hash"], "RECEIVED")
+
+    try:
+        call(
+            function="get_balance",
+            address=deploy_info["address"],
+            abi_path=ABI_PATH,
+        )
```

Comment:
After `call`, this must `raise` a RuntimeError saying that the previous line should have raised. It will only be raised if `call` itself did not raise. Like here (it seems that pytest has `pytest.fail()` which could be used with an appropriate message):
https://github.com/Shard-Labs/starknet-hardhat-example/blob/7d3381600caed8e29c2e8de1e2cda3fb04aa78d5/test/sample-test.ts#L67

Reply:
Fixed in 120646c326153f6908c609b1b257bba40b32abae